### PR TITLE
Make ObjectRef more explicit in type

### PR DESCRIPTION
**Pull request by TTWNO**
Make ObjectRef more explicit in that the "name" refers to a `zbus_names::BusName`, and not just any String.

### CODE REVIEW

File: atspi-common/src/cache.rs [hunk 1]
```diff
@@ -37,15 +37,15 @@ impl Default for CacheItem {
 	fn default() -> Self {
 		Self {
 			object: ObjectRef {
-				name: ":0.0".into(),
+				name: ":0.0".try_into().unwrap(),
```
Comment (DataTriny) [inline atspi-common/src/cache.rs:40]: I think it's acceptable to use `from_static_str` in cases like this. There's no need to bring error handling when the value is known at compile time.

**Review by DataTriny — CHANGES_REQUESTED** [review]

**Review by DataTriny — APPROVED** [review]
Looks good to me! The amount of repetitive code here is quite annoying though...

**Comment by TTWNO** [comment]
> Looks good to me! The amount of repetitive code here is quite annoying though...

Addressed by #179 , where `GenericEvent` is split up into smaller traits, which are much, much easier to implement via `macro_rules!` and blanket impls.

I've already requested a review from Luuk, but I'd appreciate a look from yourself as well.

**Comment by DataTriny** [comment]
Will try to do it tomorrow.

**Pull request merged**